src/main/java/org/candlepin/subscriptions/util/ApplicationClock.java

```diff
@@ -41,166 +31,12 @@
  * <p>All start* methods return the time at midnight - 2019-04-19 00:00:00.0 All end* methods return
  * the max time of the day - 2019-04-19 23:59:59.999999999Z
  */
-public class ApplicationClock {
+@NoArgsConstructor
+public class ApplicationClock extends CoreApplicationClock {
```

Comment:
I'd rather keep the base class named "ApplicationClock", and have the enhanced version named "ApiApplicationClock". This'll also reduce scope of changes :-)

Reply:
Moved ApplicationClock back to the base class because we'll be refactoring this more in depth when we break out which future services need which methods of the ApplicationClock